[Inductor] Fix AOT weight alignment issue on CPU

leslie-fang-intel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #135205

**Summary**
Fix issue: https://github.com/pytorch/pytorch/issues/135027. On CPU, the `consts_size` used to generate `_binary_constants_bin_start` is not padded to `ALIGN_BYTES`, while `serialized_weights` is, causing a failure in the 16K alignment check.



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @rec

Differential Revision: [D62307347](https://our.internmc.facebook.com/intern/diff/D62307347)

leslie-fang-intel committed:
[Inductor] Enable UniformValueConstantFolder for general get_attr node

[ghstack-poisoned]

jgong5 reviewed (changes requested):
CI failing?

leslie-fang-intel committed:
Update on "[Inductor] Fix AOT weight alignment issue on CPU"


**Summary**
Fix issue: https://github.com/pytorch/pytorch/issues/135027, the `consts_size` used to calculate `_binary_constants_bin_start` is not padding to `ALIGN_BYTES` but `serialized_weights` does which failed the check of `16K` alignment.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

leslie-fang-intel commented:
> CI failing?

DLRM has been marked as `expected` failure previously, changed it to pass as it should be fixed by this PR. Seems we can take this model as the UT to cover this issue.

leslie-fang-intel committed:
Update on "[Inductor] Fix AOT weight alignment issue on CPU"


**Summary**
Fix issue: https://github.com/pytorch/pytorch/issues/135027. On CPU, the `consts_size` used to calculate `_binary_constants_bin_start` is not padded to `ALIGN_BYTES`, while `serialized_weights` is, causing a failure in the 16K alignment check.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang rec

[ghstack-poisoned]

desertfire commented on torch/_inductor/codecache.py:
```diff
@@ -1728,10 +1730,23 @@ def _compile_consts_darwin(consts: bytes) -> str:
 
             output_o = os.path.splitext(input_path)[0] + ".o"
+
+            all_cuda = all(
```
I originally proposed to skip alignment for all_cuda case, but it seems unnecessarily complex. Ok to stick with it for this PR, but I will create a followup PR to skip all_cuda check, and verify if it looks fine for GPU.

leslie-fang-intel committed:
Update on "[Inductor] Fix AOT weight alignment issue on CPU"


**Summary**
Fix issue: https://github.com/pytorch/pytorch/issues/135027. On CPU, the `consts_size` used to generate `_binary_constants_bin_start` is not padded to `ALIGN_BYTES`, while `serialized_weights` is, causing a failure in the 16K alignment check.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang rec

[ghstack-poisoned]

leslie-fang-intel commented:
The same failure of `test/inductor/test_cudacodecache.py::TestCUDACodeCache::test_cuda_load` also exists on main: https://github.com/pytorch/pytorch/actions/runs/10722567567/job/29736827985

leslie-fang-intel commented:
@pytorchbot merge -i "un-related ci failure"

leslie-fang-intel commented:
@pytorchbot merge -f "un-related ci failure"

aorenste commented:
@aorenste has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62307347).
